src/article.ts

```diff
@@ -181,17 +205,36 @@ const parseElement =
 
 }
 
+type Elements = BlockElement[] | undefined;
+
 const parseElements =
-    (docParser: DocParser) => (elements: BlockElement[]): Result<string, BodyElement>[] =>
-    elements.map(parseElement(docParser));
+    (docParser: DocParser) => (elements: Elements): Result<string, BodyElement>[] => {
+        if (!elements) {
+            return [new Err('No body elements available')];
+        }
+        return elements.map(parseElement(docParser));
+    }
+
+const capiDateTimeToDate = (date: CapiDateTime | undefined | string): Option<Date> => {
+    // Thrift definitions define some dates as CapiDateTime but CAPI returns strings
+    try {
+        if (date && typeof date === 'string') {
+            return new Some(new Date(date));
+        }
+        return new None();
+    } catch(e) {
+        console.error(`Unable to convert date from CAPI: ${e}`);
```

Comment:
If we have an error message, should this return `Result<string, Date>`?

Reply:
Will that get caught later?